core/unmarshal_test.go

```diff
@@ -0,0 +1,639 @@
+	assert.Equal(t, []string{uuid1, uuid2}, actual)
+
+	slice, err = UnmarshalUUIDSlice(testMap, "not_a_slice")
+	assert.NotNil(t, err)
+	assert.Nil(t, slice)
+	assert.True(t, strings.Contains(err.Error(), "should be an array but was"))
+	t.Logf("Expected error: %s\n", err.Error())
+
+	slice, err = UnmarshalUUIDSlice(testMap, "incorrect_slice_type")
+	assert.NotNil(t, err)
+	assert.Nil(t, slice)
+	assert.True(t, strings.Contains(err.Error(), "array element should be a UUID but was"))
+	t.Logf("Expected error: %s\n", err.Error())
+
+	slice, err = UnmarshalUUIDSlice(testMap, "XXX")
+	assert.Nil(t, err)
+	assert.Nil(t, slice)
+}
+
+func TestUnmarshalDate(t *testing.T) {
+	date1 := "1970-01-01"
+	date2 := "2019-12-23"
+
+	jsonTemplate := `{
+		"prop1": "%s",
+		"slice1": ["%s","%s"],
+		"incorrect_type": true,
+		"not_a_slice": false,
+		"incorrect_slice_type": [true, false],
+		"invalid_date": "this is not a valid date",
+		"invalid_date_slice": ["another invalid date value"]
+	}`
+	jsonString := fmt.Sprintf(jsonTemplate, date1, date1, date2)
+
+	testMap, err := unmarshalJsonToMap(t, jsonString)
+	assert.Nil(t, err)
+	assert.NotNil(t, testMap)
+
+	value, err := UnmarshalDate(testMap, "prop1")
+	assert.Nil(t, err)
+	assert.NotNil(t, value)
+	assert.Equal(t, date1, value.String())
+
+	value, err = UnmarshalDate(testMap, "incorrect_type")
+	assert.NotNil(t, err)
+	assert.Nil(t, value)
+	assert.True(t, strings.Contains(err.Error(), "should be a Date but was"))
+	t.Logf("Expected error: %s\n", err.Error())
+
+	value, err = UnmarshalDate(testMap, "invalid_date")
+	assert.NotNil(t, err)
+	assert.Nil(t, value)
+	assert.True(t, strings.Contains(err.Error(), "error decoding"))
+	t.Logf("Expected error: %s\n", err.Error())
+
+	value, err = UnmarshalDate(testMap, "XXX")
+	assert.Nil(t, err)
+	assert.Nil(t, value)
+
+	slice, err := UnmarshalDateSlice(testMap, "slice1")
+	assert.Nil(t, err)
+	assert.NotNil(t, slice)
+	actual := []string{slice[0].String(), slice[1].String()}
+	assert.Equal(t, []string{date1, date2}, actual)
+
+	slice, err = UnmarshalDateSlice(testMap, "not_a_slice")
+	assert.NotNil(t, err)
+	assert.Nil(t, slice)
+	assert.True(t, strings.Contains(err.Error(), "should be an array but was"))
+	t.Logf("Expected error: %s\n", err.Error())
+
+	slice, err = UnmarshalDateSlice(testMap, "incorrect_slice_type")
+	assert.NotNil(t, err)
+	assert.Nil(t, slice)
+	assert.True(t, strings.Contains(err.Error(), "array element should be a Date but was"))
+	t.Logf("Expected error: %s\n", err.Error())
+
+	slice, err = UnmarshalDateSlice(testMap, "invalid_date_slice")
+	assert.NotNil(t, err)
+	assert.Nil(t, slice)
+	assert.True(t, strings.Contains(err.Error(), "error decoding"))
+	t.Logf("Expected error: %s\n", err.Error())
+
+	slice, err = UnmarshalDateSlice(testMap, "XXX")
+	assert.Nil(t, err)
+	assert.Nil(t, slice)
+}
+
+func TestUnmarshalDateTime(t *testing.T) {
+	datetime1 := "1970-01-01T01:02:03"
+	datetime2 := "2019-12-23T23:59:59"
```

Comment:
The [API Handbook](https://test.cloud.ibm.com/docs/api-handbook?topic=api-handbook-types#date-time)  requires date-time values to be in one of two specific formats:
- `YYYY-MM-DDTHH:mm:ssZ`
- `YYYY-MM-DDTHH:mm:ss.sssZ`

You should change this test to verify correct unmarshalling of both of these date formats.